proof/refine/RISCV64/Retype_R.thy

```diff
@@ -3685,20 +3786,21 @@ lemma createNewCaps_state_refs_of':
         apply (rename_tac apiobject_type)
         apply (case_tac apiobject_type; simp split del: if_split)
             apply (rule hoare_pre, wp, simp)
-           apply (insert cover not_0)
-           apply (wp mapM_x_wp' createObjects_state_refs_of'' threadSet_state_refs_of'
-                    | simp add: not_0 pspace_no_overlap'_def objBitsKO_def APIType_capBits_def
-                                valid_pspace'_def makeObject_tcb makeObject_endpoint objBits_def
-                                makeObject_notification archObjSize_def createObjects_def
-                                curDomain_def bit_simps
-             | intro conjI impI)+
-  done
+           apply (insert cover not_0 tysc)
+           by (wp mapM_x_wp' createObjects_state_refs_of'' threadSet_state_refs_of'
+               | simp add: not_0 pspace_no_overlap'_def objBitsKO_def APIType_capBits_def
+                           valid_pspace'_def makeObject_tcb makeObject_endpoint objBits_def
+                           makeObject_notification pageBits_def ptBits_def
+                           archObjSize_def createObjects_def curDomain_def scBits_simps
+                           pteBits_def makeObject_sc makeObject_reply pte_bits_def pt_bits_def
+                           table_size_def ptTranslationBits_def word_size_bits_def
+               | intro conjI impI )+
 
 lemma createObjects_iflive':
   "\<lbrace>\<lambda>s. if_live_then_nonz_cap' s \<and> \<not> live' val
         \<and> n \<noteq> 0
         \<and> range_cover ptr sz (objBitsKO val + gbits) n
-        \<and> pspace_aligned' s \<and> pspace_distinct' s
+        \<and> pspace_aligned' s \<and> pspace_distinct' s \<and> pspace_bounded' s
```

Comment:
for the ignorant: what does pspace_bounded represent?

Reply:
It says that the size of everything in the heap is bounded by `word_bits`. In full,

`pspace_bounded' s \<equiv>
  \<forall>x \<in> dom (ksPSpace s). objBitsKO (the (ksPSpace s x)) < word_bits`